pyhanko_certvalidator/registry.py

```diff
@@ -377,9 +377,12 @@ async def fetch_missing_potential_issuers(self, cert: x509.Certificate):
         if self.fetcher is None:
             return
 
-        async for issuer in self.fetcher.fetch_cert_issuers(cert):
-            # register the cert for future reference
-            self.register(issuer)
```

Comment:
Good catch on this one! The order indeed shouldn't matter, but I guess emitting these values eagerly possibly has some unintended side effects.

The test case you added also acts as a regression test for this one, I presume?

Reply:
I added test in the new version with a mocked `RequestsCertificateFetcher`  to cover this one.